compiler/setup.py

```diff
@@ -12,9 +14,15 @@ def readme():
     """
     return readme_short
 
+
+def quilt_version():
+    pkgdir = os.path.split(os.path.abspath(__file__))[0]
+    return open(os.path.join(pkgdir, 'VERSION')).read().strip()
```

Comment:
Need to use `with open(...):`, so the fd is not leaked.

Reply:
In what case would the FD be leaked here?

* There's no potential for a circular reference here
* The file is opened and read immediately, then closes automatically when not assigned to a variable.
* The closest possible thing to a leak would be an exception in read(), which would result in a program exit
  * Even if that were the case, the FD would only remain open (held in the stack frame associated with the traceback at sys.last_traceback) until the next exception occurs.
    * Even if another exception didn't occur (unlikely) the FD would only remain open until setup exited, and would have no ill effects.
  * Even if the OS is hard-killing the process (KILL as opposed to INT on POSIX systems), then *all* open files are generally 'leaked', and it's up to the OS to reclaim those -- in any case, that's Python and the OS's job, not ours.